service/src/main/java/org/cbioportal/genome_nexus/service/internal/VariantAnnotationServiceImpl.java

```diff
@@ -110,52 +98,32 @@ public VariantAnnotation getAnnotation(String variant, String isoformOverrideSou
     private VariantAnnotation getVariantAnnotation(String variant)
         throws VariantAnnotationNotFoundException, VariantAnnotationWebServiceException
     {
-        boolean saveAnnotationJson = true;
         VariantAnnotation variantAnnotation = null;
-        String annotationJSON = null;
+
         try {
-            variantAnnotation = variantAnnotationRepository.findOne(variant);
+            // get the annotation from the web service and save it to the DB
+            variantAnnotation = cachedExternalResourceFetcher.fetchAndCache(variant);
+
+            // include original variant value too
+            if (variantAnnotation != null) {
+                variantAnnotation.setVariant(variant);
```

Comment:
I am not sure why we needed this `variant` field. It always have the same value with the `id` field. Keeping this for backward compatibility.